src/Native/String.js

```diff
@@ -192,7 +192,7 @@ function indexes(sub, str)
 	var subLen = sub.length;
 	var i = 0;
 	var is = [];
-	while ((i = str.indexOf(sub, i)) > -1)
+	while (subLen > 0 && (i = str.indexOf(sub, i)) > -1)
```

Comment:
Wouldn't it be faster to have the check for `subLen > 0` outside the `while` loop? Seems like it only needs to happen once in an `if`, not on every loop.